### PR TITLE
chore: adding makefile to project [PS-662]

### PR DESCRIPTION
**Pull request by brad-getpassport**
Adding the makefile to the project

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+install-all: install-docs install-dev-test
+
+#
+# Clean functions
+#
+clean:
+	pyclean .
+	pre-commit clean
+	for folder in [ .mypy_cache .pytest_cache .coverage .hypothesis *.egg-info dist build ] ; do \
```
Comment (jens-getpassport) [inline Makefile:53]: What about the `docs/_build` folder?

**Review by jens-getpassport — COMMENTED** [review]
None of the `make` rules actually installs the package itself, right?

File: Makefile (hunk 1 above)
Comment (brad-getpassport) [reply Makefile:53]: Yeah I have to add that as well as fix this up.  I'll get back to it and get it done...  good catch

**Review by jens-getpassport — COMMENTED** [review]
How about we add something like
```makefile
activate:
	source venv/bin/activate
```
and then have the other rules depend on that? Somehow we need to make sure that all `pip` calls run in an activated environment, e.g.
```makefile
setup-python: activate
	...
```
Also, we could add
```makefile
docs:
	cd docs/ && make html && cd ..

check:
	pre-commit run --all-files

test:
	pytest
```
which helps us towards [SLSA](https://slsa.dev/) because these rules allow us to specify a [scripted build](https://slsa.dev/spec/v0.1/requirements#scripted-build).

**Comment by brad-getpassport** [comment]
> ```makefile
> activate:
> 	source venv/bin/activate
> ```

This can't work because the makefile is it's own shell environment and the `source` command doesn't exist.  We could add an explicit . venv/bin/activate in front of each, but making it dependent doesn't work as make fires off a separate process for each dependencies.

**Review by jens-getpassport — APPROVED** [review]
Thanks @brad-getpassport! Should we copy bits and pieces upstream into [jenstroeger/python-package-template](https://github.com/jenstroeger/python-package-template)?

> This can't work because the makefile is it's own shell environment and the `source` command doesn't exist.

Oh, it’s `bash`, and `make` uses `sh` I presume?

> […] but making it dependent doesn't work as make fires off a separate process for each dependencies.

Gah, you’re right, there’s some more discussion [here](https://stackoverflow.com/questions/27282660/setting-environment-variables-in-a-makefile) 🤓

**Comment by jens-getpassport** [comment]
If we install the Python virtual environment to `venv/` do we need to add that folder to `.gitignore`?

**Comment by brad-getpassport** [comment]
venv already in .gitignore

**Comment by jens-getpassport** [comment]
A moment too late but I’m almost tempted to use `.venv` instead of `venv` because the dotted name won’t show with `ls` (unless I ask for it). Ah well ☺️

**Pull request merged**